src/spatialdata/dataloader/datasets.py

```diff
@@ -1,188 +1,395 @@
-        Torch Dataset that returns image tiles around regions from a SpatialData object.
+        :class:`torch.utils.data.Dataset` for loading tiles from a :class:`spatialdata.SpatialData` object.
+
+        By default, the dataset returns spatialdata object, but when `return_image` and `return_annot`
+        are set, the dataset may return a tuple containing:
+
+            - the tile image, centered in the target coordinate system of the region.
+            - a vector or scala value from the table.
 
         Parameters
         ----------
         sdata
-            The SpatialData object containing the regions and images from which to extract the tiles from.
+            The :class`spatialdata.SpatialData` object.
         regions_to_images
-            A dictionary mapping the regions element key we want to extract the tiles around to the images element key
-            we want to get the image data from.
+            A mapping betwen region and images. The regions are used to compute the tile centers, while the images are
+            used to get the pixel values.
+        regions_to_coordinate_systems
+            A mapping between regions and coordinate systems. The coordinate systems are used to transform both
+            regions coordinates for tiles as well as images.
+        tile_scale
+            The scale of the tiles. This is used only if the `regions` are `shapes`.
+            It is a scaling factor applied to either the radius (spots) or length (polygons) of the `shapes`
+            according to the geometry type of the `shapes` element:
+
+                - if `shapes` are circles (spots), the radius is scaled by `tile_scale`.
+                - if `shapes` are polygons, the length of the polygon is scaled by `tile_scale`.
+
+            If `tile_dim_in_units` is passed, `tile_scale` is ignored.
         tile_dim_in_units
             The dimension of the requested tile in the units of the target coordinate system. This specifies the extent
             of the image each tile is querying. This is not related he size in pixel of each returned tile.
-        tile_dim_in_pixels
-            The dimension of the requested tile in pixels. This specifies the size of the output tiles that we will get,
-             independently of which extent of the image the tile is covering.
-        target_coordinate_system
-            The coordinate system in which the tile_dim_in_units is specified.
+        raster
+            If True, the images are rasterized using :func:`spatialdata.rasterize`.
+            If False, they are rasterized using :func:`spatialdata.bounding_box_query`.
+        return_annot
+            If not None, a value from the table is returned together with the image tile.
+            Only columns in :attr:`anndata.AnnData.obs` and :attr:`anndata.AnnData.X`
+            can be returned. If None, it will return a spatialdata object with only the tuple
+            containing the image and the table value.
```

Comment:
I took a look at that function and not sure would fit well what I'm doing there, as in it's passing views of the annotations instead of copies of the full tables. I think I'd have to modify it in order to use it in that context. I understand that current implementation is also not great though

Reply:
You are right. I'd keep the current implementation then since it's still quite simple, we can always change things later if needed.